packages/server/src/routes.ts

```diff
@@ -113,6 +114,10 @@ export const createRouter = (ctx: AppContext): express.Router => {
   // Update or create a DID doc
   router.post('/:did', async function (req, res) {
     const { did } = req.params
+    // TODO: Provide carve-out for big clients like Bluesky
+    await rateLimitPerHour(ctx, res, 'did-post-ip', req.ip, 1000)
+    await rateLimitPerDay(ctx, res, 'did-post-day', did, 30)
+    await rateLimitPerHour(ctx, res, 'did-post-hour', did, 10)
```

Comment:
one thing that we didn't talk about but that occurred to me is: we actually only want to rate limit _successful_ operations

(well we might want to limit successful ops too cause they take work on our side, but those could be much higher)

I believe the library has a way to either peek the limit & consume at the end of execution or consume at the start & refund on failure

thoughts?

Reply:
I could see a potential failure scenario where there is a bug in some other code that eats up all the retries for the day with bad requests, and then even if the code is fixed it still requires a full 24h to resolve since it can't process the requests. That would be a pretty bad failure mode, actually. I'll take a look and see what can be done